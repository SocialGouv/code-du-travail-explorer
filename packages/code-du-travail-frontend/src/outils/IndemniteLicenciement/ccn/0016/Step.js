import React from "react";
import { Categorie, CADRE } from "./Categorie";
import { CategoriePeriod } from "./CategoriePeriod";
import { YesNoQuestion } from "../../../common/YesNoQuestion";
import { TextQuestion } from "../../../common/TextQuestion";

function Step({ form }) {
  const state = form.getState();
  const { anciennete = 0, age = 0, categorie = "" } = state.values;
  const askDetailedCategories = anciennete >= 3 && categorie === CADRE;

  return (
    <>
      <Categorie name="categorie" />
      {askDetailedCategories && <CategoriePeriod />}
      <TextQuestion
        inputType="number"
        name="age"
        size="5"
        label="Quel etait votre âge à la date du licenciement&nbsp;?"
      />
      {age > 60 && (
        <YesNoQuestion
          label="Le salarié pourrait-il bénéficier de la retraite&nbsp;?"
          name="hasRetirementAge"
        />
      )}
    </>
  );
}

export { Step };
