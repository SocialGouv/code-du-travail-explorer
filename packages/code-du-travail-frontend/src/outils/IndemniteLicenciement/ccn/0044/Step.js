import { Title } from "@socialgouv/cdtn-react-ui";
import React from "react";
import { Field } from "react-final-form";

import { TextQuestion } from "../../../common/TextQuestion";
import { YesNoQuestion } from "../../../common/YesNoQuestion";
import { EchelonChimie } from "./EchelonChimie";

function Step() {
  return (
    <>
      <Title as="h3">Chimie</Title>
      <EchelonChimie name="groupe" />

      <Field name="anciennete">
        {({ input }) => {
          const anciennete = Math.floor(input.value);
          if ((anciennete >= 1 && anciennete < 2) || anciennete > 5) {
            return (
              <>
                <YesNoQuestion
                  name="hasOpe"
                  label="Est-ce que l'entreprise est affiliée à une des organisations patronales employeur&nbsp;?"
                />
                <YesNoQuestion
                  name="isEco"
                  label="Le motif du licenciement est-il économique ?"
                />
                {anciennete > 5 && (
                  <TextQuestion
                    inputType="number"
                    name="age"
                    label="Quel était l’âge du salarié à la date du licenciement ?"
                  />
                )}
              </>
            );
          }
          return null;
        }}
      </Field>
    </>
  );
}

export { Step };
