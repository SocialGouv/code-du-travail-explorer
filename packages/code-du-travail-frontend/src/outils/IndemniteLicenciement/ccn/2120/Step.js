import React from "react";
import { Field } from "react-final-form";
import styled from "styled-components";
import { theme } from "@socialgouv/react-ui";

import { SectionTitle, Label } from "../../../common/stepStyles";

export const DISCIPLINAIRE = "disciplinaire";
export const NON_DISCIPLINAIRE = "non-disciplinaire";
export const ECONOMIQUE = "eco";

export const CADRE = "cadre";
export const NON_CADRE = "non-cadre";
export const NE_SAIT_PAS = "ne-sais-pas";

export const optionMotifs = {
  [DISCIPLINAIRE]: "Motif disciplinaire",
  [NON_DISCIPLINAIRE]: "Motif non disciplinaire",
  [ECONOMIQUE]: "Motif économique"
};

export const optionCategorie = {
  [NON_CADRE]: "Non cadre",
  [CADRE]: "Cadre",
  [NE_SAIT_PAS]: "Ne sait pas"
};

function Step() {
  return (
    <>
      <SectionTitle>Motif du licenciement</SectionTitle>
      <Field name="motif">
        {({ input }) => {
          return (
            <FormGroup>
              <Label htmlFor="input-motif">
                Quel est le motif du licenciement&nbsp;?
              </Label>
              <Select id="input-motif" {...input}>
                <option value="" disabled>
                  motifs...
                </option>
                {Object.entries(optionMotifs).map(([value, label]) => (
                  <option key={value} value={value}>
                    {label}
                  </option>
                ))}
              </Select>
            </FormGroup>
          );
        }}
      </Field>

      <Field name="categorie">
        {({ input }) => {
          return (
            <FormGroup>
              <Label htmlFor="input-categorie">
                Quelle est la catégorie professionnelle du salarié&nbsp;?
              </Label>

              <Select id="input-categorie" {...input}>
                <option value="" disabled>
                  catégories...
                </option>
                {Object.entries(optionCategorie).map(([value, label]) => (
                  <option key={value} value={value}>
                    {label}
                  </option>
                ))}
              </Select>
            </FormGroup>
          );
        }}
      </Field>
    </>
  );
}

export { Step };

const { spacings } = theme;
const Select = styled.select`
  max-width: 32rem;
`;

const FormGroup = styled.div`
  display: flex;
  flex-direction: column;
  align-items: flex-start;
  margin-bottom: ${spacings.medium};
`;
