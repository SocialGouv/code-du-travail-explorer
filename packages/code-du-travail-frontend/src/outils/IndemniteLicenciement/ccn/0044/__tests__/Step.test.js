import { render } from "@testing-library/react";
import React from "react";
import { Form } from "react-final-form";

import { Step } from "../Step";

describe("<Step />", () => {
  it("should render", () => {
    const onSubmit = jest.fn();
    const { container } = render(
      <Form onSubmit={onSubmit} render={() => <Step />} />
    );
    expect(container).toMatchSnapshot();
  });
  it("should ask for ope and licenciement eco and age", () => {
    const onSubmit = jest.fn();
    const { container } = render(
      <Form
        onSubmit={onSubmit}
        initialValues={{ anciennete: 6 }}
        render={() => <Step />}
      />
    );
    expect(container).toMatchSnapshot();
  });
  it("should ask for ope and licenciement eco", () => {
    const onSubmit = jest.fn();
    const { container } = render(
      <Form
        onSubmit={onSubmit}
        initialValues={{ anciennete: 1.5 }}
        render={() => <Step />}
      />
    );
    expect(container).toMatchSnapshot();
  });
});
