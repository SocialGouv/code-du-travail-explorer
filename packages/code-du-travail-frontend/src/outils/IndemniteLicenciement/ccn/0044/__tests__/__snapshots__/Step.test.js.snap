// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Step /> should ask for ope and licenciement eco 1`] = `
.c0 {
  margin: 0 0 1rem 0;
  color: #2f3b6c;
  font-weight: normal;
  font-size: 2.8rem;
  font-family: "Merriweather",serif;
  line-height: 1.25;
}

.c2 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  margin-right: 2em;
  padding: 0;
}

.c5 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-align-items: flex-start;
  -webkit-box-align: flex-start;
  -ms-flex-align: flex-start;
  align-items: flex-start;
  -webkit-box-pack: start;
  -webkit-justify-content: flex-start;
  -ms-flex-pack: start;
  justify-content: flex-start;
  margin-bottom: 2rem;
}

.c1 {
  display: block;
  margin-top: 2rem;
  margin-bottom: 1rem;
  font-size: 1.8rem;
  cursor: default;
}

.c4 {
  display: inline-block;
  margin-left: 1rem;
  color: #eb5757;
  font-weight: 700;
}

.c3 {
  width: 100%;
}

<div>
  <h3
    class="c0"
  >
    Chimie
  </h3>
  <p
    class="c1"
  >
    Quel est l’échelon du salarié dans la convention collective ?
    <br />
    Cette information se trouve sur le bulletin de salaire.
  </p>
  <label
    class="c2"
    for="echelon"
  >
    Sélectionnez un échelon
  </label>
  <select
    class="c3"
    id="echelon"
    name="echelon"
  >
    <option
      disabled=""
      value=""
    />
    <option
      value="130"
    >
      130
    </option>
    <option
      value="140"
    >
      140
    </option>
    <option
      value="150"
    >
      150
    </option>
    <option
      value="160"
    >
      160
    </option>
    <option
      value="175"
    >
      175
    </option>
    <option
      value="190"
    >
      190
    </option>
    <option
      value="205"
    >
      205
    </option>
    <option
      value="225"
    >
      225
    </option>
    <option
      value="235"
    >
      235
    </option>
    <option
      value="250"
    >
      250
    </option>
    <option
      value="275"
    >
      275
    </option>
    <option
      value="300"
    >
      300
    </option>
    <option
      value="325"
    >
      325
    </option>
    <option
      value="350"
    >
      350
    </option>
    <option
      value="360"
    >
      360
    </option>
    <option
      value="400"
    >
      400
    </option>
    <option
      value="460"
    >
      460
    </option>
    <option
      value="480"
    >
      480
    </option>
    <option
      value="510"
    >
      510
    </option>
    <option
      value="550"
    >
      550
    </option>
    <option
      value="660"
    >
      660
    </option>
    <option
      value="770"
    >
      770
    </option>
    <option
      value="880"
    >
      880
    </option>
  </select>
  <p
    class="c1"
  >
    Est-ce que l'entreprise est affiliée à une des organisations patronales employeur ?
    <span
      aria-label="champs obligatoire"
      class="c4"
    >
      *
    </span>
  </p>
  <div
    class="c5"
  >
    <label
      class="c2"
    >
      <input
        name="hasOpe"
        type="radio"
        value="true"
      />
      <span>
        Oui
      </span>
    </label>
    <label
      class="c2"
    >
      <input
        name="hasOpe"
        type="radio"
        value="false"
      />
      <span>
        Non
      </span>
    </label>
  </div>
  <p
    class="c1"
  >
    Le motif du licenciement est-il économique ?
    <span
      aria-label="champs obligatoire"
      class="c4"
    >
      *
    </span>
  </p>
  <div
    class="c5"
  >
    <label
      class="c2"
    >
      <input
        name="isEco"
        type="radio"
        value="true"
      />
      <span>
        Oui
      </span>
    </label>
    <label
      class="c2"
    >
      <input
        name="isEco"
        type="radio"
        value="false"
      />
      <span>
        Non
      </span>
    </label>
  </div>
</div>
`;

exports[`<Step /> should ask for ope and licenciement eco and age 1`] = `
.c0 {
  margin: 0 0 1rem 0;
  color: #2f3b6c;
  font-weight: normal;
  font-size: 2.8rem;
  font-family: "Merriweather",serif;
  line-height: 1.25;
}

.c8 {
  width: 10em;
  text-align: right;
  border-color: #bbcadf;
}

.c8::-webkit-outer-spin-button,
.c8::-webkit-inner-spin-button {
  margin: 0;
  -webkit-appearance: none;
  -moz-appearance: none;
  appearance: none;
}

.c2 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  margin-right: 2em;
  padding: 0;
}

.c5 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-align-items: flex-start;
  -webkit-box-align: flex-start;
  -ms-flex-align: flex-start;
  align-items: flex-start;
  -webkit-box-pack: start;
  -webkit-justify-content: flex-start;
  -ms-flex-pack: start;
  justify-content: flex-start;
  margin-bottom: 2rem;
}

.c1 {
  display: block;
  margin-top: 2rem;
  margin-bottom: 1rem;
  font-size: 1.8rem;
  cursor: default;
}

.c6 {
  display: block;
  margin-top: 2rem;
  margin-bottom: 1rem;
  font-size: 1.8rem;
  cursor: pointer;
}

.c4 {
  display: inline-block;
  margin-left: 1rem;
  color: #eb5757;
  font-weight: 700;
}

.c3 {
  width: 100%;
}

.c7 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  margin-bottom: 2rem;
}

<div>
  <h3
    class="c0"
  >
    Chimie
  </h3>
  <p
    class="c1"
  >
    Quel est l’échelon du salarié dans la convention collective ?
    <br />
    Cette information se trouve sur le bulletin de salaire.
  </p>
  <label
    class="c2"
    for="echelon"
  >
    Sélectionnez un échelon
  </label>
  <select
    class="c3"
    id="echelon"
    name="echelon"
  >
    <option
      disabled=""
      value=""
    />
    <option
      value="130"
    >
      130
    </option>
    <option
      value="140"
    >
      140
    </option>
    <option
      value="150"
    >
      150
    </option>
    <option
      value="160"
    >
      160
    </option>
    <option
      value="175"
    >
      175
    </option>
    <option
      value="190"
    >
      190
    </option>
    <option
      value="205"
    >
      205
    </option>
    <option
      value="225"
    >
      225
    </option>
    <option
      value="235"
    >
      235
    </option>
    <option
      value="250"
    >
      250
    </option>
    <option
      value="275"
    >
      275
    </option>
    <option
      value="300"
    >
      300
    </option>
    <option
      value="325"
    >
      325
    </option>
    <option
      value="350"
    >
      350
    </option>
    <option
      value="360"
    >
      360
    </option>
    <option
      value="400"
    >
      400
    </option>
    <option
      value="460"
    >
      460
    </option>
    <option
      value="480"
    >
      480
    </option>
    <option
      value="510"
    >
      510
    </option>
    <option
      value="550"
    >
      550
    </option>
    <option
      value="660"
    >
      660
    </option>
    <option
      value="770"
    >
      770
    </option>
    <option
      value="880"
    >
      880
    </option>
  </select>
  <p
    class="c1"
  >
    Est-ce que l'entreprise est affiliée à une des organisations patronales employeur ?
    <span
      aria-label="champs obligatoire"
      class="c4"
    >
      *
    </span>
  </p>
  <div
    class="c5"
  >
    <label
      class="c2"
    >
      <input
        name="hasOpe"
        type="radio"
        value="true"
      />
      <span>
        Oui
      </span>
    </label>
    <label
      class="c2"
    >
      <input
        name="hasOpe"
        type="radio"
        value="false"
      />
      <span>
        Non
      </span>
    </label>
  </div>
  <p
    class="c1"
  >
    Le motif du licenciement est-il économique ?
    <span
      aria-label="champs obligatoire"
      class="c4"
    >
      *
    </span>
  </p>
  <div
    class="c5"
  >
    <label
      class="c2"
    >
      <input
        name="isEco"
        type="radio"
        value="true"
      />
      <span>
        Oui
      </span>
    </label>
    <label
      class="c2"
    >
      <input
        name="isEco"
        type="radio"
        value="false"
      />
      <span>
        Non
      </span>
    </label>
  </div>
  <label
    class="c6"
    for="1"
  >
    Quel était l’âge du salarié à la date du licenciement ?
    <span
      aria-label="champs obligatoire"
      class="c4"
    >
      *
    </span>
  </label>
  <div
    class="c7"
  >
    <input
      class="c8"
      id="1"
      name="age"
      type="number"
      value=""
    />
  </div>
</div>
`;

exports[`<Step /> should render 1`] = `
.c0 {
  margin: 0 0 1rem 0;
  color: #2f3b6c;
  font-weight: normal;
  font-size: 2.8rem;
  font-family: "Merriweather",serif;
  line-height: 1.25;
}

.c2 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  margin-right: 2em;
  padding: 0;
}

.c1 {
  display: block;
  margin-top: 2rem;
  margin-bottom: 1rem;
  font-size: 1.8rem;
  cursor: default;
}

.c3 {
  width: 100%;
}

<div>
  <h3
    class="c0"
  >
    Chimie
  </h3>
  <p
    class="c1"
  >
    Quel est l’échelon du salarié dans la convention collective ?
    <br />
    Cette information se trouve sur le bulletin de salaire.
  </p>
  <label
    class="c2"
    for="echelon"
  >
    Sélectionnez un échelon
  </label>
  <select
    class="c3"
    id="echelon"
    name="echelon"
  >
    <option
      disabled=""
      value=""
    />
    <option
      value="130"
    >
      130
    </option>
    <option
      value="140"
    >
      140
    </option>
    <option
      value="150"
    >
      150
    </option>
    <option
      value="160"
    >
      160
    </option>
    <option
      value="175"
    >
      175
    </option>
    <option
      value="190"
    >
      190
    </option>
    <option
      value="205"
    >
      205
    </option>
    <option
      value="225"
    >
      225
    </option>
    <option
      value="235"
    >
      235
    </option>
    <option
      value="250"
    >
      250
    </option>
    <option
      value="275"
    >
      275
    </option>
    <option
      value="300"
    >
      300
    </option>
    <option
      value="325"
    >
      325
    </option>
    <option
      value="350"
    >
      350
    </option>
    <option
      value="360"
    >
      360
    </option>
    <option
      value="400"
    >
      400
    </option>
    <option
      value="460"
    >
      460
    </option>
    <option
      value="480"
    >
      480
    </option>
    <option
      value="510"
    >
      510
    </option>
    <option
      value="550"
    >
      550
    </option>
    <option
      value="660"
    >
      660
    </option>
    <option
      value="770"
    >
      770
    </option>
    <option
      value="880"
    >
      880
    </option>
  </select>
</div>
`;
